ui/core/src/model/query.ts

```diff
@@ -12,16 +12,35 @@
 // limitations under the License.
 
 import { Definition, UnknownSpec } from './definitions';
+import { TimeSeriesData } from './time-series-data';
 
 interface QuerySpec<PluginSpec> {
   plugin: Definition<PluginSpec>;
 }
 /**
  * A generic query definition interface that can be extended to support more than just TimeSeriesQuery
  */
-export interface QueryDefinition<Kind = QueryType, PluginSpec = UnknownSpec> {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface QueryDefinition<Kind = any, PluginSpec = UnknownSpec> {
   kind: Kind;
   spec: QuerySpec<PluginSpec>;
 }
 
-export type QueryType = 'TimeSeriesQuery';
+/**
+ * Mapping the query plugin to the data type it returns
+ */
+export interface QueryType {
+  TimeSeriesQuery: TimeSeriesData;
+  // in the future we can add other query plugin and data types
+  // LogsQuery: LogsData;
+}
+
+// export interface LogsData {
```

Comment:
Could you clarify in these comments that these are here as examples for future dev (as opposed to real implementations that were commented out for some other reason). I'm hoping that'll help reduce confusion in the future.